test/units/dump-scalar-styles.js

```diff
@@ -3,79 +3,234 @@
+      assert.equal(yaml.safeDump('42'), '\'42\'\n');
+      assert.equal(yaml.safeDump('99.9'), '\'99.9\'\n');
+      assert.equal(yaml.safeDump('127.0001'), '\'127.0001\'\n');
+      assert.equal(yaml.safeDump('1.23015e+3'), '\'1.23015e+3\'\n');
+    });
+
+    test('quote leading/trailing whitespace', function () {
+      assert.equal(yaml.safeDump(' leading space'), '\' leading space\'\n');
+      assert.equal(yaml.safeDump('trailing space '), '\'trailing space \'\n');
+    });
+
+    test('quote leading quotes', function () {
+      assert.equal(yaml.safeDump("'singles double'"), "'''singles double'''\n");
+      assert.equal(yaml.safeDump('"single double'), '\'"single double\'\n');
+    });
+
+    test('escape non-printables', function () {
+      assert.equal(yaml.safeDump('a\nb\u0001c'), '"a\\nb\\x01c"\n');
+    });
+  });
 
-test('fold', function () {
-  var essay = '';
-  for (var i = 1001; i < 1200; i++) {
-    if (i % 50 === 0) {
-      essay += '\n';
-    } else if (essay) {
-      if (i % 11 === 0 || i === 1066) {
-        // throw some double-spaces in there for good measure.
-        // this forces 1065 and 1066 to the same line.
-        essay += '  ';
-      } else {
-        essay += ' ';
+  suite('Literal style', function () {
+    var content = 'a\nb \n\n c\n  d', indented = indent(content);
+
+    test('preserves trailing newlines using chomping', function () {
+      assert.equal(yaml.safeDump({ a: '\n', b: '\n\n', c: 'c\n', d: 'd\nd' }),
+        'a: |+\n\nb: |+\n\n\nc: |\n  c\nd: |-\n  d\n  d\n');
+      assert.equal(yaml.safeDump('\n'),               '|+\n' + '\n');
+      assert.equal(yaml.safeDump('\n\n'),             '|+\n' + '\n\n');
+
+      assert.equal(yaml.safeDump(content),            '|-\n' + indented + '\n');
+      assert.equal(yaml.safeDump(content + '\n'),     '|\n'  + indented + '\n');
+      assert.equal(yaml.safeDump(content + '\n\n'),   '|+\n' + indented + '\n\n');
+      assert.equal(yaml.safeDump(content + '\n\n\n'), '|+\n' + indented + '\n\n\n');
+    });
+
+    test('accepts leading whitespace', function () {
+      assert.equal(yaml.safeDump('   ' + content), '|2-\n   ' + indented + '\n');
+    });
+
+    test('falls back to quoting when required indent indicator is too large', function () {
+      assert.equal(yaml.safeDump(' these go\nup to\neleven', { indent: 11 }),
+        '" these go\\nup to\\neleven"\n');
+    });
+
+    test('does not use block style for multiline key', function () {
+      assert.equal(yaml.safeDump({
+        'push\nand': {
+          you: 'pull'
+        }
+      }), '"push\\nand":\n  you: pull\n');
+    });
+  });
+
+  suite('Folded style', function () {
+    (function () {
+      var content = (function () {
+        var result = '';
+        var i = 1000;
+        for (var para = 1; para <= 7; para++) {
+          result += '\n';
+          // indent paragraphs 3 and 4
+          if (para === 3 || para === 4) {
+            result += ' '.repeat(para);
```

Comment:
`.repeat()` is poorly supported. Needs helper or `Array(para).join(' ')` to fix tests.

Run tests under node 0.10 for sure.